apps/journeys-admin/src/components/Editor/ControlPanel/Attributes/blocks/Footer/HostSidePanel/HostSidePanel.spec.tsx

```diff
@@ -0,0 +1,317 @@
+      firstName: 'User',
+      id: user1.id,
+      imageUrl: 'imageURL',
+      lastName: '1'
+    }
+  }
+
+  const defaultHost = {
+    id: 'hostId',
+    __typename: 'Host',
+    teamId: userTeam.id,
+    title: 'Cru International',
+    location: 'Florida, USA',
+    src1: 'imageSrc1',
+    src2: 'imageSrc2'
+  }
+
+  const journey = {
+    __typename: 'Journey',
+    id: 'journeyId',
+    seoTitle: 'My awesome journey',
+    host: defaultHost,
+    team: { id: userTeam.id, title: 'My team' }
+  } as unknown as Journey
+
+  const getUserTeamMock: MockedResponse<GetUserTeamsAndInvites> = {
+    request: {
+      query: GET_USER_TEAMS_AND_INVITES,
+      variables: {
+        teamId: userTeam.id,
+        where: { role: [UserTeamRole.manager, UserTeamRole.member] }
+      }
+    },
+    result: {
+      data: {
+        userTeams: [userTeam],
+        userTeamInvites: []
+      }
+    }
+  }
+
+  it('should render the default host side panel', async () => {
+    const { getAllByRole } = render(
+      <MockedProvider mocks={[getUserTeamMock]}>
+        <ThemeProvider>
+          <JourneyProvider
+            value={{
+              journey: { ...journey, host: null },
+              variant: 'admin'
+            }}
+          >
+            <HostSidePanel />
+          </JourneyProvider>
+        </ThemeProvider>
+      </MockedProvider>
+    )
+
+    await waitFor(() => {
+      expect(
+        getAllByRole('button', { name: 'Select a Host' })[0]
+      ).not.toBeDisabled()
+    })
+  })
+
+  it('should disable editing hosts if no team on journey', async () => {
+    const { getAllByRole } = render(
+      <MockedProvider mocks={[getUserTeamMock]}>
+        <ThemeProvider>
+          <JourneyProvider
+            value={{
+              journey: { ...journey, team: null, host: null },
+              variant: 'admin'
+            }}
+          >
+            <HostSidePanel />
+          </JourneyProvider>
+        </ThemeProvider>
+      </MockedProvider>
+    )
+
+    await waitFor(() => {
+      expect(
+        getAllByRole('button', { name: 'Select a Host' })[0]
+      ).toBeDisabled()
+    })
+  })
+
+  it('should disable editing hosts if current user does not have access', async () => {
+    useCurrentUserMock.mockReturnValue({
+      loadUser: jest.fn(),
+      data: { id: 'otherUser', email: 'otherUser@test.com' }
+    })
+
+    const { getAllByRole, getAllByText } = render(
+      <MockedProvider mocks={[getUserTeamMock]}>
+        <ThemeProvider>
+          <JourneyProvider
+            value={{
+              journey: { ...journey, host: null },
+              variant: 'admin'
+            }}
+          >
+            <HostSidePanel />
+          </JourneyProvider>
+        </ThemeProvider>
+      </MockedProvider>
+    )
+
+    await waitFor(() => {
+      expect(
+        getAllByRole('button', { name: 'Select a Host' })[0]
+      ).toBeDisabled()
+      expect(
+        getAllByText(`Only My team members can edit this`)[0]
```

Comment:
Is it possible to use the team name variable in here? Looks like the intention was already there since you used the back ticks?
```suggestion
        getAllByText(`Only ${My} team members can edit this`)[0]
```

Reply:
Yeah, I originally used the variable but since journey.team.title can be null it didn't like it. Will remove back ticks.